### PR TITLE
Add configurable MAX_FORKS for jobs

### PR DESCRIPTION
**Pull request by jakemcdermott**
Add configurable MAX_FORKS for jobs

##### SUMMARY
Add a configurable limit for job forks
see #5142 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API
 - UI

##### ADDITIONAL INFORMATION
settings
Screenshot from 2020-01-06 18-42-58

save
max_forks_01

launch
Screenshot from 2020-01-15 13-15-26

### CODE REVIEW

**Review by ryanpetrello — COMMENTED** [review]
Should we add something that enforces this after-the-fact (maybe in the `tasks.py` code that looks up the actual forks value)?  i.e.,

- User makes a JT w/ 5000 forks
- Admin notices and sets the max to 20
- User runs the JT

In this scenario, I think the job should still probably run, but maybe we enforce 20, and print a loud warning in the logs.

**Comment by jakemcdermott** [comment]
> In this scenario, I think the job should still probably run, but maybe we enforce 20, and print a loud warning in the logs.

Sounds good to me. We can log a warning instead of completely preventing the launch. 

By enforce, do you mean just log the warning or log the warning _and_ stop the job before the playbook gets executed?

**Comment by ryanpetrello** [comment]
I'm of the opinion that we shouldn't stop the job from running - just cap the forks with `min(job.forks, max_allowed_forks)` and print a warning in the logs.

File: awx/main/tasks.py [hunk 1]
```diff
@@ -1656,8 +1656,12 @@ def build_args(self, job, private_data_dir, passwords):
                     args.append('--vault-id')
                     args.append('{}@prompt'.format(vault_id))
 
-        if job.forks:  # FIXME: Max limit?
-            args.append('--forks=%d' % job.forks)
+        if job.forks:
+            if settings.MAX_FORKS > 0 and job.forks > settings.MAX_FORKS:
+                logger.warning(f'Maximum number of forks ({settings.MAX_FORKS}) exceeded.')
```
Comment (ryanpetrello) [inline awx/main/tasks.py:1661]: 👍

**Review by ryanpetrello — APPROVED** [review]
Looks good to me - thanks!

File: awx/main/tasks.py (hunk 1 above)
Comment (jakemcdermott) [inline awx/main/tasks.py:1661]: Screenshot from 2020-01-15 13-15-26

File: awx/main/tests/functional/api/test_job_template.py [hunk 2]
```diff
@@ -118,6 +118,21 @@ def test_extra_credential_unique_type_xfail(get, post, organization_factory, job
...
+    inventory.use_role.members.add(alice)
+    settings.MAX_FORKS = 10
+    response = post(reverse('api:job_template_list'), {
+        'name': 'Some name',
+        'project': project.id,
+        'inventory': inventory.id,
+        'playbook': 'helloworld.yml',
+        'forks': 11,
+    }, alice)
+    assert response.status_code == 400
```
Comment (AlanCoding) [inline awx/main/tests/functional/api/test_job_template.py:133]: job templates have a high chance of hitting 400 errors, so I want to make sure this hits the one we want

```python
assert 'Maximum number of forks (10) exceeded' in str(response.data)
```

or something like that

**Comment by AlanCoding** [comment]
made one small test comment

otherwise no objections from me

File: awx/main/tests/functional/api/test_job_template.py (hunk 2 above)
Comment (jakemcdermott) [reply awx/main/tests/functional/api/test_job_template.py:133]: @AlanCoding good idea. It is done.

**Pull request merged**